src/zope/interface/_compat.py

```diff
@@ -56,3 +62,104 @@ def _skip_under_py3k(test_method):
 def _skip_under_py2(test_method):
     import unittest
     return unittest.skipIf(sys.version_info[0] < 3, "Only on Python 3")(test_method)
+
+
+def _c_optimizations_required():
+    """
+    Return a true value if the C optimizations are required.
+
+    This uses the ``PURE_PYTHON`` variable as documented in `_use_c_impl`.
+    """
+    pure_env = os.environ.get('PURE_PYTHON')
+    require_c = pure_env == "0"
+    return require_c
+
+
+def _c_optimizations_available():
+    """
+    Return the C optimization module, if available, otherwise
+    a false value.
+
+    If the optimizations are required but not available, this
+    raises the ImportError.
+
+    This does not say whether they should be used or not.
+    """
+    catch = () if _c_optimizations_required() else (ImportError,)
+    try:
+        from zope.interface import _zope_interface_coptimizations as c_opt
+        return c_opt
+    except catch: # pragma: no cover (only Jython doesn't build extensions)
+        return False
+
+
+def _c_optimizations_ignored():
+    """
+    The opposite of `_c_optimizations_required`.
+    """
+    pure_env = os.environ.get('PURE_PYTHON')
+    return pure_env is not None and pure_env != "0"
+
+
+def _should_attempt_c_optimizations():
+    """
+    Return a true value if we should attempt to use the C optimizations.
+
+    This takes into account whether we're on PyPy and the value of the
+    ``PURE_PYTHON`` environment variable, as defined in `_use_c_impl`.
+    """
+    is_pypy = hasattr(sys, 'pypy_version_info')
+
+    if _c_optimizations_required():
+        return True
+    if is_pypy:
+        return False
+    return not _c_optimizations_ignored()
+
+def _use_c_impl(py_impl, name=None, globs=None):
```

Comment:
PEP-8: two blank lines between functions.

Reply:
And I was being so careful about function spacing, too. I even corrected it several other places! 😉